components/device-mgt/org.wso2.carbon.device.mgt.api/src/test/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceTypeManagementServiceTest.java

```diff
@@ -0,0 +1,245 @@
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.FeatureManager;
+import org.wso2.carbon.device.mgt.common.push.notification.PushNotificationConfig;
+import org.wso2.carbon.device.mgt.common.type.mgt.DeviceTypeMetaDefinition;
+import org.wso2.carbon.device.mgt.core.dto.DeviceType;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderServiceImpl;
+import org.wso2.carbon.device.mgt.jaxrs.service.api.DeviceTypeManagementService;
+import org.wso2.carbon.device.mgt.jaxrs.service.impl.util.DeviceMgtAPITestUtils;
+import org.wso2.carbon.device.mgt.jaxrs.util.DeviceMgtAPIUtils;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.ws.rs.core.Response;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+/**
+ * This class holds the unit tests for the class {@link DeviceTypeManagementService}
+ */
+@PowerMockIgnore("javax.ws.rs.*")
+@SuppressStaticInitializationFor({"org.wso2.carbon.device.mgt.jaxrs.util.DeviceMgtAPIUtils",
+        "org.wso2.carbon.context.CarbonContext"})
+@PrepareForTest({DeviceMgtAPIUtils.class, MultitenantUtils.class, CarbonContext.class,
+        DeviceManagementProviderService.class})
+public class DeviceTypeManagementServiceTest {
+
+    private static final Log log = LogFactory.getLog(DeviceManagementServiceImplTest.class);
+    private static final String TEST_DEVICE_TYPE = "TEST-DEVICE-TYPE";
+    private static final int TEST_DEVICE_TYPE_ID = 12345;
+    private static final String DEVICE_TYPE_DESCRIPTION = "TEST DESCRIPTION";
+    private static final String MODIFIED_SINCE = "1234503934242";
+    private DeviceTypeManagementService deviceTypeManagementService;
+    private DeviceManagementProviderService deviceManagementProviderService;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeClass
+    public void init() throws DeviceManagementException {
+        log.info("Initializing DeviceTypeManagement tests");
+        initMocks(this);
+        this.deviceManagementProviderService = Mockito
+                .mock(DeviceManagementProviderServiceImpl.class, Mockito.RETURNS_MOCKS);
+        this.deviceTypeManagementService = new DeviceTypeManagementServiceImpl();
+    }
+
+    @Test(description = "Testing for existing device types.")
+    public void testExistingDeviceType() throws Exception {
+        PowerMockito.stub(PowerMockito.method(DeviceMgtAPIUtils.class, "getDeviceManagementService"))
+                .toReturn(this.deviceManagementProviderService);
+        Response response = this.deviceTypeManagementService.getDeviceTypes("");
+        Assert.assertNotNull(response);
```

Comment:
Shall we add a next parameter, to specify why the test case is failing? Can you update in all the test cases